cts/cts-exec.in

```diff
@@ -847,19 +850,21 @@ class ExecTests(Tests):
 
         # Verify agents only exist in their lists
         test = self.new_test("verify_agent_lists", "Verify the agent lists contain the right data.")
-        test.add_cmd(args="-c list_agents ", stdout_match="Stateful")                                          # ocf
-        test.add_cmd(args="-c list_agents -C ocf", stdout_match="Stateful")
-        test.add_cmd(args="-c list_agents -C lsb", stdout_match="", stdout_no_match="Stateful")                # should not exist
-        test.add_cmd(args="-c list_agents -C service", stdout_match="", stdout_no_match="Stateful")            # should not exist
-        test.add_cmd(args="-c list_agents ", stdout_match="LSBDummy")                                          # init.d
-        test.add_cmd(args="-c list_agents -C lsb", stdout_match="LSBDummy")
-        test.add_cmd(args="-c list_agents -C service", stdout_match="LSBDummy")
-        test.add_cmd(args="-c list_agents -C ocf", stdout_match="", stdout_no_match="pacemaker-cts-dummyd@")   # should not exist
-
-        test.add_cmd(args="-c list_agents -C ocf", stdout_match="", stdout_no_match="pacemaker-cts-dummyd@")   # should not exist
-        test.add_cmd(args="-c list_agents -C lsb", stdout_match="", stdout_no_match="fence_dummy")             # should not exist
-        test.add_cmd(args="-c list_agents -C service", stdout_match="", stdout_no_match="fence_dummy")         # should not exist
-        test.add_cmd(args="-c list_agents -C ocf", stdout_match="", stdout_no_match="fence_dummy")             # should not exist
+
+        if "ocf" in self._rsc_classes:
+            test.add_cmd(args="-c list_agents ", stdout_match="Stateful")
+            test.add_cmd(args="-c list_agents -C ocf", stdout_match="Stateful",
+                         stdout_no_match="(pacemaker-cts-dummyd@|fence_dummy)")
+
+        if "service" in self._rsc_classes:
+            test.add_cmd(args="-c list_agents -C service", stdout_match="",
+                         stdout_no_match="(Stateful|fence_dummy)")
+
+        if "lsb" in self._rsc_classes:
+            test.add_cmd(args="-c list_agents", stdout_match="LSBDummy")
+            test.add_cmd(args="-c list_agents -C lsb", stdout_match="LSBDummy",
+                         stdout_no_match="(pacemaker-cts-dummyd@|Stateful|fence_dummy)")
```

Comment:
I don't think you need parens in any of the stdout_no_match regexes.  I think they'll get used for capturing submatches so they can be used as a reference later (like `\1`, etc.) which we don't need to do.